### DISCUSSION TITLE
Creating an empty TrueType file (.ttf)

### DISCUSSION THREAD

**Discussion by EzBlueMN** [fonttools/fonttools#3178] [Q&A] [answered]
I've been experiencing with ChatGPT to find a way to simply create an empty font and save it so I could later on insert some glyph from another source (.ttf). But I can't manage to make a simple file without different kind of error. The error are coming from outside my code and the interaction with TTFont...
This is the first suggestion which isn't even compatible
```
   def create_empty(self, font_path=None, family_name='New Family'):
        # Create an empty TTFont object
        ttfont = TTFont()

        # Add required tables to the font object
        ttfont['head'] = TTFont.newTable('head')
        ttfont['hhea'] = TTFont.newTable('hhea')
        ttfont['maxp'] = TTFont.newTable('maxp')
        ttfont['post'] = TTFont.newTable('post')
        ttfont['name'] = TTFont.newTable('name')

        ttfont['name'].setName(family_name, 1, 1, 0, 0)

        # Save the font to a file
        ttfont.save(font_path)
```
AttributeError: type object 'TTFont' has no attribute 'newTable'

or

```
    def create_empty(self, font_path=None, family_name='New Family'):
        # Create an empty TTFont object
        ttfont = TTFont()

        # Add required tables to the font object
        ttfont['head'] = ttLib.newTable('head')
        ttfont['hhea'] = ttLib.newTable('hhea')
        ttfont['maxp'] = ttLib.newTable('maxp')
        ttfont['post'] = ttLib.newTable('post')
        ttfont['name'] = ttLib.newTable('name')

        ttfont['name'].setName(family_name, 1, 1, 0, 0)

        # Save the font to a file
        ttfont.save(font_path)
```
fontTools.ttLib.TTLibError: illegal use of getGlyphOrder()
<class 'fontTools.ttLib.TTLibError'>

I can't seems to find a way to simply have an empty font
The ultimate goal would be to be able to copy front a source to a destination font file with the possibility of changing the unicode value that correspond to the glyph or to change the glyph_name...
Thanks

**Comment by ftCLI** [comment, marked as answer]
Here you can find the instructions on how to build a minimal TTF font: https://github.com/fonttools/fonttools/blob/060b5f36bf46094ae2fd511e2544aad7624fbdc8/Lib/fontTools/fontBuilder.py#L25

**Reply by EzBlueMN** [reply]
Thank you,
Right what I needed!
